src/onnx/onnx.cpp

```diff
@@ -325,16 +325,32 @@ struct onnx_parser
                                const attribute_map& attributes,
                                std::vector<instruction_ref> args)
     {
-        op::clip op;
+        auto input_lens = args[0]->get_shape().lens();
+        instruction_ref min_arg;
```

Comment:
If the operator has only one input and there is not "min" or "max", the min_arg and max_arg will be empty here. The logic would be something like 
        if(args.size() > 2)
        {
            max_arg = args[2];
        }
        else
        {
            float max_val = std::numeric_limits<float>::max();
            if(contains(attributes, "max"))
            {
                max_val = parse_value(attributes.at("max")).at<float>();
            }
            max_arg = prog.add_literal(max_val);
        }

        if(args.size() > 1)
        {
            min_arg = args[1];
        }
        else
        {
            float min_val = std::numeric_limits<float>::min();
            if(contains(attributes, "min"))
            {
                min_val = parse_value(attributes.at("min")).at<float>();
            }
            min_arg = prog.add_literal(min_val);
        }

If there is no min or max value provided, the default value is used. (Actually, for this case, the clip operator can be removed.)

Reply:
I agree that I need to fix the logic. I am also debating on whether to assign default values to min/max or simply using a max operator when no max provided, and an identity if neither operators are provided.

Reply:
Yes, I am fine with not adding the operator if neither of min/max is used.

Reply:
fixed

Reply:
fixed